src/apis/emscripten/env.rs

```diff
@@ -148,6 +143,49 @@ pub extern "C" fn _getgrnam(name_ptr: c_int, instance: &mut Instance) -> c_int {
     }
 }
 
+pub fn call_malloc(size: i32, instance: &mut Instance) -> u32 {
+    let ret = instance
+        .call("_malloc", &[Value::I32(size)])
+        .expect("_malloc call failed");
+    if let Some(Value::I32(ptr)) = ret {
+        ptr as u32
+    } else {
+        panic!("unexpected value from _malloc: {:?}", ret);
+    }
+}
+
+pub fn call_memalign(alignment: u32, size: u32, instance: &mut Instance) -> u32 {
+    let ret = instance
+        .call(
+            "_memalign",
+            &[Value::I32(alignment as i32), Value::I32(size as i32)],
+        )
+        .expect("_memalign call failed");
+    if let Some(Value::I32(res)) = ret {
+        res as u32
+    } else {
+        panic!("unexpected value from _memalign {:?}", ret);
+    }
+}
+
+pub fn call_memset(pointer: u32, value: i32, size: u32, instance: &mut Instance) -> u32 {
+    let ret = instance
+        .call(
+            "_memset",
+            &[
+                Value::I32(pointer as i32),
+                Value::I32(value),
+                Value::I32(size as i32),
+            ],
+        )
+        .expect("_memset call failed");
+    if let Some(Value::I32(res)) = ret {
+        res as u32
+    } else {
+        panic!("unexpected value from _memset {:?}", ret);
+    }
+}
```

Comment:
These new utility functions are very generic. Do we have a shared place for useful tools like this?

Reply:
Yes, maybe this belongs in `utils` or maybe a new `commons` module.